fire: init at 1.0.0.3

OPNA2608 opened the pull request:
###### Description of changes

Packages [Fire](https://github.com/jerryuhoo/Fire), a multi-band VST3 & AU distortion plugin.

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

OPNA2608 commented on pkgs/applications/audio/fire/default.nix:
```diff
@@ -0,0 +1,105 @@
+  };
+
+  patches = [
+    ./0001-Remove-FetchContent-usage.patch
```
Upstream is currently mixing Git submodules & CMake FetchContent, I've opened an issue asking them to decide on one (preferably submodules so we can just include them with `fetchSubmodules`): https://github.com/jerryuhoo/Fire/issues/46

OPNA2608 committed:
fire: init at 0.9.9

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/prs-ready-for-review/3032/1206

SuperSandro2000 commented on pkgs/applications/audio/fire/default.nix:
```diff
@@ -0,0 +1,105 @@
+    description = "Multi-band distortion plugin by Wings";
+    homepage = "https://github.com/jerryuhoo/Fire";
+    license = licenses.agpl3Only; # Not clarified if Only or Plus
+    platforms = platforms.all;
```
```suggestion
    platforms = platforms.unix;
```

OPNA2608 replied:
`all` isn't accurate cuz of all the embedded targets in there which have no use for VST3s, but neither is just `unix` if we're honest… meh.

SuperSandro2000 commented on pkgs/applications/audio/fire/default.nix:
```diff
@@ -0,0 +1,105 @@
+
+  doCheck = stdenv.buildPlatform.canExecute stdenv.hostPlatform;
+
+  # JUCE handles installation during buildPhase
```
Could we split that or is that none trivial to do?

OPNA2608 replied:
It's abit messier IMO, but done.

SuperSandro2000 commented on pkgs/applications/audio/fire/default.nix:
```diff
@@ -0,0 +1,105 @@
+    CoreAudioKit
+  ];
+
+  NIX_CFLAGS_COMPILE = lib.optionalString stdenv.hostPlatform.isDarwin (toString [
+    # Fails to find fp.h on its own
+    "-isystem ${CoreServices}/Library/Frameworks/CoreServices.framework/Versions/Current/Frameworks/CarbonCore.framework/Versions/Current/Headers/"
+  ]);
```
```suggestion
  # Fails to find fp.h on its own
  NIX_CFLAGS_COMPILE = lib.optionalString stdenv.hostPlatform.isDarwin "-isystem ${CoreServices}/Library/Frameworks/CoreServices.framework/Versions/Current/Frameworks/CarbonCore.framework/Versions/Current/Headers/";
```

SuperSandro2000 reviewed (commented):
otherwise LGTM

OPNA2608 committed:
fire: PR feedback

OPNA2608 committed:
fire: 0.9.9 -> 1.0.0.3

SuperSandro2000 commented on pkgs/top-level/all-packages.nix:
```diff
@@ -27532,6 +27532,10 @@ with pkgs;
 
   filezilla = callPackage ../applications/networking/ftp/filezilla { };
 
+  fire = callPackage ../applications/audio/fire {
+    inherit (darwin.apple_sdk.frameworks) Cocoa WebKit CoreServices DiscRecording CoreAudioKit;
```
```suggestion
  fire = darwin.apple_sdk_11_0.callPackage ../applications/audio/fire {
    inherit (darwin.apple_sdk_11_0.frameworks) Cocoa WebKit CoreServices DiscRecording CoreAudioKit;
```
maybe that fixes the darwin build failure already

OPNA2608 replied:
Also needed some additional dependencies, fixed now.

OPNA2608 committed:
fire: Fix on Darwin after bump

Needs at least 10.14 SDK on Darwin, so we need to switch to the 11.0 one.

https://github.com/juce-framework/JUCE/blob/b08520c2de1771af3dfcbfbc0e0b6b0b5eb083b0/modules/juce_core/system/juce_TargetPlatform.h#L147-L148

gador commented:
Result of `nixpkgs-review pr 188515` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>fire</li>
  </ul>
</details>
